packages/agents/src/agents/utils/Agent.ts

```diff
@@ -36,11 +36,21 @@ export class Agent<TRunArgs = GoalRunArgs> implements RunnableAgent<TRunArgs> {
     return this.context.workspace;
   }
 
-  public async *run(
-    args: TRunArgs
-  ): AsyncGenerator<AgentOutput, RunResult, string | undefined> {
-    await this.initializeChat(args);
+  public async init() {
+    await this.context.chat.persistent([
+      { role: "system", content: `Variables are annotated using the \${variable-name} syntax. Variables can be used as function argument using the \${variable-name} syntax. Variables are created as needed, and do not exist unless otherwise stated.` },
+      ...this.config.prompts.initialMessages()
+    ]);
+  }
 
+  public async *run(args: TRunArgs): AsyncGenerator<AgentOutput, RunResult, string | undefined> {
+    await this.init();
+    return yield* this.runWithExistingContext(args);
+  }
+
+  public async *runWithExistingContext(args: TRunArgs): AsyncGenerator<AgentOutput, RunResult, string | undefined> {
+    await this.initRun(args);
```

Comment:
nit: now we have `run`, `init` and `initRun` and it seems a bit confusing

if I understand correctly the purpose of each is:
`run`: it initializes the chat and runs with existing context
`init`: adds the initial messages to chat
`initRun`: adds the goal as a persistent message

would it make sense to maybe name the `initRun` something like `addGoal`? feels more readable imo